src/core/labeling/qgspallabeling.h

```diff
@@ -475,6 +475,12 @@ class CORE_EXPORT QgsPalLayerSettings
      */
     bool prepare( QgsRenderContext &context, QSet<QString> &attributeNames SIP_INOUT, const QgsFields &fields, const QgsMapSettings &mapSettings, const QgsCoordinateReferenceSystem &crs );
 
+    /**
+     * Returns all field names referenced by the configuration (e.g. field name or expression, data defined properties).
+     * \since QGIS 3.14
+     */
+    QSet<QString> referencedFields() const;
```

Comment:
I think this method should require a QgsExpressionContext argument

Reply:
Could you please explain this a bit? My understanding was that we need QgsExpressionContext only if we want to also "prepare" the object - e.g. figure out field index from QgsFields. But if we only need a list of field names, we shouldn't need that...?

Reply:
MMm.... it's going back a long time here. I think from memory it was something to do with ambiguous field names which aren't quoted in the expressions, and needing to know in advance the available set of fields in order to resolve this ambiguity. But it's a long time ago, and I might be confusing issues!

My comment today is because every other classes requires an expression context or render context in their equivalent of referencedFields (unhelpfully named a variety of referencedFields/usedAttributes/referencedColumns/... !), so we should keep that consistency so at least the behavior is predictable across all the different classes.

Reply:
Yeah now I recall something like that too - I think you are right - in old times we were only guessing if a string was a field name or actual expression, to make that e.g. "my-field" is correctly - based on whether it was present as a QgsField. But nowadays as far as I know thanks to your work on QgsProperty we should not have this problem anymore.

In this particular case, if I add QgsExpressionContext, I will also need to add another "ignoreContext" argument and call the function always with empty context and ignoreContext=true which would be ugly. And we do not have much consistency elsewhere either - e.g. QgsExpression::referencedColumns() has no arguments, and callouts/symbols want QgsRenderContext :exploding_head:   So if you don't mind I would prefer to keep it as it is... (this whole referencedFields/usedAttributes/referencedColumns stuff will need refactoring in qgis 4.0)